docs/INSTALL.md

```diff
@@ -0,0 +1,101 @@
+# Setup Ecko
+
+There are currently a few ways to install Ecko for a production environment.
+
+1. [YunoHost](https://install-app.yunohost.org/?app=ecko) - easiest, recommended for beginners and small instances
+
+2. Docker-compose - medium-difficulty, for those familiar with docker - [doc](./INSTALL_DOCKER_COMPOSE.md)
+
+3. Installing from Source - larger or customized instances, see below
+
+## Installing from Source
+
+Tested on Debian 11 Bullseye
+
+### Install Node and Yarn
+
+    curl -sL https://deb.nodesource.com/setup_12.x | bash -
+
+And as suggested by the above script
+
+    sudo apt-get install -y nodejs
+    curl -sL https://dl.yarnpkg.com/debian/pubkey.gpg | gpg --dearmor | sudo tee /usr/share/keyrings/yarnkey.gpg >/dev/null
+    echo "deb [signed-by=/usr/share/keyrings/yarnkey.gpg] https://dl.yarnpkg.com/debian stable main" | sudo tee /etc/apt/sources.list.d/yarn.list
+    sudo apt-get update && sudo apt-get install yarn
+
+### Install system dependencies
+    apt install -y \
+      imagemagick ffmpeg libpq-dev libxml2-dev libxslt1-dev file git-core \
+      g++ libprotobuf-dev protobuf-compiler pkg-config nodejs gcc autoconf \
+      bison build-essential libssl-dev libyaml-dev libreadline6-dev \
+      zlib1g-dev libncurses5-dev libffi-dev libgdbm-dev \
+      nginx redis-server redis-tools postgresql postgresql-contrib \
+      certbot python3-certbot-nginx yarn libidn11-dev libicu-dev libjemalloc-dev
+
+### Setup user
+    adduser --disabled-login mastodon
+    su - mastodon
+
+### Install Ruby and bundler via rbenv
```

Comment:
There can be rvm installs as well since rbenv sometimes skips versions. Anyways it works :)